assets/tests/components/mapPage.test.tsx

```diff
@@ -1,72 +1,59 @@
-import React from "react"
+import "@testing-library/jest-dom"
 import { render, screen } from "@testing-library/react"
+import React from "react"
 import { BrowserRouter } from "react-router-dom"
-import "@testing-library/jest-dom"
 import MapPage from "../../src/components/mapPage"
 import { StateDispatchProvider } from "../../src/contexts/stateDispatchContext"
 import useSearchResults from "../../src/hooks/useSearchResults"
-import { Ghost, Vehicle, VehicleOrGhost } from "../../src/realtime"
-import { initialState } from "../../src/state"
 import * as dateTime from "../../src/util/dateTime"
 
-import vehicleFactory from "../factories/vehicle"
-import ghostFactory from "../factories/ghost"
 import userEvent from "@testing-library/user-event"
 import { useTripShape } from "../../src/hooks/useShapes"
-import { SearchPageState } from "../../src/state/searchPageState"
 import useVehicleForId from "../../src/hooks/useVehicleForId"
+import { SearchPageState } from "../../src/state/searchPageState"
+import stateFactory from "../factories/applicationState"
+import ghostFactory from "../factories/ghost"
+import { RunFactory } from "../factories/run"
+import { searchPageStateFactory } from "../factories/searchPageState"
+import { searchQueryRunFactory } from "../factories/searchQuery"
+import shapeFactory from "../factories/shape"
+import vehicleFactory from "../factories/vehicle"
+
 jest
   .spyOn(dateTime, "now")
   .mockImplementation(() => new Date("2018-08-15T17:41:21.000Z"))
 
 jest.spyOn(Date, "now").mockImplementation(() => 234000)
 
-const vehicle: Vehicle = vehicleFactory.build()
-
-const ghost: Ghost = ghostFactory.build({
-  id: "ghost-trip",
-  directionId: 0,
-  routeId: "39",
-  tripId: "trip",
-  headsign: "headsign",
-  blockId: "block",
-  runId: "123-0123",
-  viaVariant: "X",
-  layoverDepartureTime: null,
-  scheduledTimepointStatus: {
-    timepointId: "t0",
-    fractionUntilTimepoint: 0.0,
-  },
-  scheduledLogonTime: null,
-  routeStatus: "on_route",
-  blockWaivers: [],
-})
-
 jest.mock("../../src/hooks/useSearchResults", () => ({
   __esModule: true,
   default: jest.fn(),
 }))
 
 jest.mock("../../src/hooks/useShapes", () => ({
   __esModule: true,
-  useTripShape: jest.fn(() => null),
+  useTripShape: jest.fn(),
 }))
 
 jest.mock("../../src/hooks/useNearestIntersection", () => ({
   __esModule: true,
-  useNearestIntersection: jest.fn(() => null),
+  useNearestIntersection: jest.fn(),
 }))
 
 jest.mock("../../src/hooks/useVehicleForId", () => ({
   __esModule: true,
   default: jest.fn(),
 }))
 
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
 describe("MapPage", () => {
   test("renders the empty state", () => {
-    ;(useSearchResults as jest.Mock).mockImplementationOnce(() => null)
+    ;(useSearchResults as jest.Mock).mockReturnValue([])
```

Comment:
**suggestion:** I think there should be test cases for both `null` and `[]` because they're both values that `useSearchResults` can return. Apologies if there is still a test for `null` and I'm just not seeing it.